lightning/src/chain/keysinterface.rs

```diff
@@ -217,6 +225,7 @@ pub trait ChannelKeys : Send+Clone {
 	/// Create a signature for a local commitment transaction. This will only ever be called with
 	/// the same local_commitment_tx (or a copy thereof), though there are currently no guarantees
 	/// that it will not be called multiple times.
+	/// The commitment must not have been revoked.
```

Comment:
Did you say this policy enforcement has been implemented, namely "don't sign a revoked transaction" ? I think you need to verify the new `revoked_commitment` against commitment transaction number. I guess you can access it by unveiling it from `LocalCommitmentTransaction` 's unsigned_tx. This can be addressed later.

Reply:
Oops, I misspoke above. I thought about implementing this, but didn't get to it.

Reply:
Maybe since it's not implemented yet, rephrase the line to be a TODO?

Reply:
Agreed, but again, I think it should be attached to `EnforcingChannelKeys`

Reply:
done

Reply:
I believe all of our other comments are phrased as "what rust-lightning guarantees it will do" ie, instead of "The commitment must not have been revoked" this could read "The commitment transaction contained will not have been revoked by release_commitment_secret, and external signers may wish to enforce this".

Reply:
reworded, please take a look